### PR TITLE
Add ability to pass smartblock cmd to REPEAT to iterate over

### PR DESCRIPTION
**Pull request by mdroidian**
Exposes these variables during the loop
- `smartBlocksContext.variables["ITERATION"] = (i + 1).toString();`
- `smartBlocksContext.variables["ITERATIONVALUE"] = values[i] || "";`

~~iteration value comes from last argument in `<%REPEAT%>`~~
`<%REPEAT:<%CHILDREN:((someUid))%>,<%SMARTBLOCK:runMe%>`

### CODE REVIEW

File: src/utils/core.ts [hunk 1]
```diff
@@ -1730,25 +1730,20 @@ export const COMMANDS: {
...
-                    return [...p, c];
-                  })
-                ),
-              Promise.resolve([] as InputTextNode[][])
-            )
-        )
-        .then((s) => s.flatMap((c) => c)),
+    delayArgs: [true, true, false],
+    help: "Repeats the current block a specified number of times\n\n1. Number of times for repeat",
+    handler: async (repeatArg = "1", content = "", ...values) => {
```
Comment (mdroidian) [inline src/utils/core.ts:1735]: settings `repeatArg` to `false` was overriding `content`

eg: `<%REPEAT:<%CHILDREN:((uid)),<%SMARTBLOCK:runMe%>%>` resulted in `content` = `child2`

File: src/utils/core.ts [hunk 2]
```diff
@@ -2277,34 +2273,41 @@ const processBlockTextToPromises = (s: string) => {
...
+    const processArgs = (
+      args: string[],
+      delayArgs: boolean[] | boolean
+    ): Promise<InputTextNode[][]> => {
+      const delayArgsArray =
+        typeof delayArgs === "boolean"
+          ? Array(args.length).fill(delayArgs)
+          : delayArgs;
+
+      const argPromises = args.map((arg, i) => {
```
Comment (dvargas92495) [inline src/utils/core.ts:2288]: blocking - change to `reduce`

File: src/utils/core.ts [hunk 3]
```diff
@@ -1730,25 +1730,21 @@ export const COMMANDS: {
...
+      const repeatCount =
+        repeatArg === "-" ? values.length : Number(repeatArgText) || 1;
+      const results = [];
+      for (let i = 0; i < repeatCount; i++) {
+        smartBlocksContext.variables["ITERATION"] = (i + 1).toString();
+        smartBlocksContext.variables["ITERATIONVALUE"] = values[i] || "";
+        const result = await proccessBlockText(content);
+        results.push(result);
+      }
+      return results.flatMap((c) => c);
```
Comment (dvargas92495) [inline src/utils/core.ts:1746]: same thing as `flatMap()`

Comment (mdroidian) [reply src/utils/core.ts:1746]: did you mean `flat()`?

File: src/utils/core.ts (hunk 1 above)
Comment (mdroidian) [reply src/utils/core.ts:1735]: This is still the case, even changing `.map` to `.reduce` @dvargas92495 

https://www.loom.com/share/07aa41eb1b48496586b7c9a4e61f6f5d

File: docs/050-command-reference.md [hunk 4]
```diff
@@ -983,14 +983,25 @@ If it isn't, the block is not inserted
...
+1. Count of repeats (or a SmartBlock command that returns items to iterate over)
 2. Content to repeat
 
 **Example**:
 
 - `<%REPEAT:5,hello%>`
+- `<%REPEAT:<%CHILDREN:((someUid))%>,<%SMARTBLOCK:runMe%>%>`
+- runMe #SmartBlock
+  - `<%GET:ITERATION%>`
+  - `<%GET:ITERATIONVALUE%>`
```
Comment (mdroidian) [inline docs/050-command-reference.md:1004]: these are the relevant doc changes, the others are just some syntax fixing

**Comment by mdroidian** [comment]
`delayArgs` required additional changes to be complete, so I decided to postpone those changes as they were no longer relevant for this PR.  And I wanted to finish this PR for this sprint.

**Review by dvargas92495 — APPROVED** [review]

**Pull request merged**